components/board/genericlinux/board.go

```diff
@@ -86,43 +76,127 @@ func (b *Board) Reconfigure(
 	_ resource.Dependencies,
 	conf resource.Config,
 ) error {
-	newConf, err := b.convertConfig(conf)
+	newConfPtr, err := b.convertConfig(conf)
 	if err != nil {
 		return err
 	}
+	newConf := *newConfPtr
```

Comment:
In your approach, you get a type error because the `*(...)` is operating on a tuple and that doesn't make sense.

I had also tried `*newConf, err := b.convertConfig(conf)`, but that has syntax errors because `*newConf` doesn't look like the name of a new variable (same thing if I try `&newConf` instead).

Arguably, the "right" way to do this is to change all the functions that use `newConf` to take in a pointer instead of the struct itself. This was changed in https://github.com/viamrobotics/rdk/pull/2729, and it looks like I did it wrong. I should just put back all the pointers in the function signatures. Thanks for calling this out!